File: apps/ios/mynewsapp/UI/ArticleDetailView.swift
import SwiftUI

struct ArticleDetailView: View {
    let viewModel: ArticleDetailViewModel
    @State private var isShowingSafari = false
    
    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 16) {
                if let urlToImage = viewModel.article.urlToImage, let url = URL(string: urlToImage) {
                    AsyncImage(url: url) { phase in
                        switch phase {
                        case .empty:
                            ProgressView()
                                .frame(height: 200)
                                .frame(maxWidth: .infinity)
                        case .success(let image):
                            image
                                .resizable()
                                .aspectRatio(contentMode: .fit)
                        case .failure:
                            EmptyView()
                        @unknown default:
                            EmptyView()
                        }
                    }
                    .frame(maxHeight: 300)
                }
                
                VStack(alignment: .leading, spacing: 8) {
                    Text(viewModel.article.title)
                        .font(.title)
                        .bold()
                    
                    if let author = viewModel.article.author {
                        Text(String(format: NSLocalizedString("author_prefix", comment: "Author prefix"), author))
                            .font(.subheadline)
                            .foregroundStyle(.secondary)
                    }
                    
                    Text(viewModel.article.publishedAtString)
                        .font(.caption)
                        .foregroundStyle(.secondary)
                    
                    Divider()
                    
                    if let description = viewModel.article.description {
                        Text(description)
                            .font(.body)
                    }
                    
                    if let content = viewModel.article.content {
                        Text(content)
                            .font(.body)
                            .padding(.top, 4)
                    }
                    
                    if let url = URL(string: viewModel.article.url) {
                        Button(NSLocalizedString("read_full_article", comment: "Read full article")) {
                            isShowingSafari = true
                        }
                        .font(.headline)
                        .padding(.top)
                        .fullScreenCover(isPresented: $isShowingSafari) {
                            SafariView(url: url)
                        }
                    }
                }
                .padding()
            }
        }
        .navigationBarTitleDisplayMode(.inline)
    }
}

File: apps/android/app/src/main/java/com/pouriaalmassi/mynewsapp/ui/ArticleDetailView.kt
package com.pouriaalmassi.mynewsapp.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import coil.compose.AsyncImage
import coil.request.ImageRequest
import com.pouriaalmassi.mynewsapp.R
import com.pouriaalmassi.mynewsapp.models.Article

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ArticleDetailView(
    article: Article,
    onReadFullArticleClick: (String) -> Unit,
    onBackClick: () -> Unit
) {
    val scrollState = rememberScrollState()

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("", maxLines = 1) }, // Emptier title matches the inline title style of iOS detail view
                navigationIcon = {
                    IconButton(onClick = onBackClick) {
                        Icon(
                            imageVector = Icons.Default.ArrowBack,
                            contentDescription = "Back"
                        )
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.background,
                    navigationIconContentColor = MaterialTheme.colorScheme.onBackground
                )
            )
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .background(MaterialTheme.colorScheme.background)
                .verticalScroll(scrollState)
        ) {
            // Header Image using Coil
            article.urlToImage?.let { imageUrl ->
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(240.dp)
                        .padding(horizontal = 16.dp, vertical = 8.dp)
                        .clip(RoundedCornerShape(16.dp))
                ) {
                    AsyncImage(
                        model = ImageRequest.Builder(LocalContext.current)
                            .data(imageUrl)
                            .crossfade(true)
                            .placeholder(R.drawable.ic_launcher_background)
                            .error(R.drawable.ic_launcher_background)
                            .build(),
                        contentDescription = article.title,
                        contentScale = ContentScale.Crop,
                        modifier = Modifier.fillMaxSize()
                    )
                }
            }

            // Article Contents
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(20.dp),
                verticalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                // Title
                Text(
                    text = article.title,
                    style = MaterialTheme.typography.titleLarge,
                    color = MaterialTheme.colorScheme.onBackground,
                    fontWeight = FontWeight.Bold
                )

                // Metadata Column (Author & Date)
                Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
                    article.author?.let { author ->
                        Text(
                            text = stringResource(id = R.string.author_prefix, author),
                            style = MaterialTheme.typography.bodyMedium,
                            color = MaterialTheme.colorScheme.secondary,
                            fontWeight = FontWeight.SemiBold
                        )
                    }

                    Text(
                        text = article.publishedAtString,
                        style = MaterialTheme.typography.labelMedium,
                        color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.5f)
                    )
                }

                Divider(
                    thickness = 1.dp,
                    color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.1f)
                )

                // Description Block
                article.description?.let { desc ->
                    Text(
                        text = desc,
                        style = MaterialTheme.typography.bodyLarge,
                        color = MaterialTheme.colorScheme.onBackground,
                        fontWeight = FontWeight.Medium
                    )
                }

                // Content Block
                article.content?.let { body ->
                    Text(
                        text = body,
                        style = MaterialTheme.typography.bodyLarge,
                        color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.8f)
                    )
                }

                Spacer(modifier = Modifier.height(10.dp))

                // CTA Button (Full Article Reader Trigger)
                Button(
                    onClick = { onReadFullArticleClick(article.url) },
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(50.dp),
                    shape = RoundedCornerShape(12.dp),
                    colors = ButtonDefaults.buttonColors(
                        containerColor = MaterialTheme.colorScheme.primary
                    )
                ) {
                    Text(
                        text = stringResource(id = R.string.read_full_article),
                        style = MaterialTheme.typography.bodyLarge,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.onPrimary
                    )
                }
            }
        }
    }
}
